app/controllers/miq_policy_controller/policies.rb

```diff
@@ -1,91 +1,113 @@
+    if @policy && @policy.id
+      add_flash(_("Edit of %{model} \"%{name}\" was cancelled by the user") % {:model => ui_lookup(:model => "MiqPolicy"), :name => @policy.description})
+    else
+      add_flash(_("Add of new %{models} was cancelled by the user") % {:models => ui_lookup(:model => "MiqPolicy")})
+    end
+    @edit = nil
+    get_node_info(x_node)
+    replace_right_cell(:nodetype => @nodetype)
+  end
+
+  def policy_edit_reset
+    @sb[:action] = "policy_edit"
+    policy_build_edit_screen(session[:edit].try(:key?, :typ) ? session[:edit][:typ] : params[:typ])
+    if params[:button] == "reset"
+      add_flash(_("All changes have been reset"), :warning)
+    end
+    replace_right_cell(:nodetype => "p")
+  end
+
+  def policy_edit_save
+    assert_privileges("policy_#{@policy.id ? "edit" : "new"}")
+    policy = @policy.id.blank? ? MiqPolicy.new : MiqPolicy.find(@policy.id) # Get new or existing record
+    policy.mode = @edit[:new][:mode]
+    policy.updated_by = session[:userid]
+
+    if @policy.id.blank?
+      policy.towhat = @edit[:new][:towhat]
+      policy.created_by = session[:userid]
+    end
+
+    case @edit[:typ]
+    when "basic"
+      policy.description = @edit[:new][:description]
+      policy.active = @edit[:new][:active]
+      policy.notes = @edit[:new][:notes]
+      policy.expression = @edit[:new][:expression]["???"] ? nil : MiqExpression.new(@edit[:new][:expression])
+    when "conditions"
+      mems = @edit[:new][:conditions].invert                  # Get the ids from the member list box
+      policy.conditions.collect { |pc| pc }.each { |c| policy.conditions.delete(c) unless mems.keys.include?(c.id) }  # Remove any conditions no longer in members
+      mems.each_key { |m| policy.conditions.push(Condition.find(m)) unless policy.conditions.collect(&:id).include?(m) }    # Add any new conditions
+    end
+
+    if !policy.valid? || @flash_array || !policy.save
+      policy.errors.each do |field, msg|
+        add_flash("#{field.to_s.capitalize} #{msg}", :error)
       end
-      @edit = nil
-      get_node_info(x_node)
-      replace_right_cell(:nodetype => @nodetype)
+      javascript_flash
       return
-    when "reset", nil # Reset or first time in
-      @sb[:action] = "policy_edit"
-      policy_build_edit_screen(session[:edit].try(:key?, :typ) ? session[:edit][:typ] : params[:typ])
-      if params[:button] == "reset"
-        add_flash(_("All changes have been reset"), :warning)
+    end
+
+    if @policy.id.blank? && policy.mode == "compliance"   # New compliance policy
+      event = MiqEventDefinition.find_by(:name => "#{policy.towhat.downcase}_compliance_check") # Get the compliance event record
+      policy.sync_events([event])                           # Set the default compliance event in the policy
+      action_list = [[MiqAction.find_by_name("compliance_failed"), {:qualifier => :failure, :synchronous => true}]]
```

Comment:
i think you should be using find_by(:name..) instead

Reply:
I agree. Actually I am fiding these issues but most of the time in separate PRs that deal with a file at a time or so.
https://github.com/ManageIQ/manageiq-ui-classic/pull/2201
https://github.com/ManageIQ/manageiq-ui-classic/pull/2197